src/value.rs

```diff
@@ -113,6 +120,99 @@ impl<'gc> Value<'gc> {
         }
     }
 
+    // A small helper function to handle division-like zero handling
+    fn safe_div<T: PartialEq + ToPrimitive + Zero>(
+        lhs: T,
+        rhs: T,
+        f: &Fn(T, T) -> Value<'gc>,
+    ) -> Value<'gc> {
+        match (lhs, rhs) {
+            // Seems that all nans are negative in lua
+            (ref a, ref b) if a.is_zero() && b.is_zero() => Value::Number(-f64::NAN),
+            (ref a, ref b) if b.is_zero() => {
+                Value::Number(copysign(f64::INFINITY, a.to_f64().unwrap()))
+            }
+            (a, b) => f(a, b),
+        }
+    }
+
+    // This operation always returns a Number, even when called by int arguments
+    pub fn float_divide(self, other: Value<'gc>) -> Option<Value<'gc>> {
+        let (a, b) = match (self, other) {
+            (Value::Integer(a), Value::Integer(b)) => (a as f64, b as f64),
+            (Value::Number(a), Value::Number(b)) => (a, b),
+            (Value::Integer(a), Value::Number(b)) => (a as f64, b),
+            (Value::Number(a), Value::Integer(b)) => (a, b as f64),
```

Comment:
It seems like there should be some `as_number` Value cast which returns an `Option<f64>`... or something better than this, there seem to be a lot of these as f64 castas and checking whether both sides can be expressed as numbers.

Reply:
Yes, I was thinking that, it would also help with String coercion down the line.  I just wasn't sure if now was the correct time to start that.

Reply:
This as been addressed by the functions `to_number`, `to_integer` and `bin_op`.